core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -990,24 +990,20 @@ class KafkaApis(val requestChannel: RequestChannel,
         val bandwidthThrottleTimeMs = quotas.fetch.maybeRecordAndGetThrottleTimeMs(request, responseSize, timeMs)
 
         val maxThrottleTimeMs = math.max(bandwidthThrottleTimeMs, requestThrottleTimeMs)
+        unconvertedFetchResponse = fetchContext.updateAndGenerateResponseData(partitions)
```

Comment:
- We also need to update FetchSessionHandler in 3.0. The [if statement here](https://github.com/linkedin/kafka/blob/42e6fae5a14c00d2947e772212a0dd9c189987f5/clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java#L413) expect empty responseData for throttling checking. With [this line](https://github.com/linkedin/kafka/pull/514/files#diff-cc056b4960ededba37a438b1454f0f3c5ff5e8ad5e6d2ec9a08e813ca056ffebR993), it will break the assumption.

I also checked for the [maybeThrottle](https://github.com/linkedin/kafka/blob/42e6fae5a14c00d2947e772212a0dd9c189987f5/clients/src/main/java/org/apache/kafka/clients/NetworkClient.java#L1356) method. It won't be impacted.